config/kubemark.yaml

```diff
@@ -0,0 +1,80 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: deleteunreadynodes
+  namespace: kube-system
+data:
+  entrypoint.sh: |-
+    #!/bin/bash
+    while true; do
+      echo "Checking NotReady nodes"
+      for node in $(kubectl get nodes -o json | jq '.items[].metadata.name' --raw-output); do
+        echo "Checking node $node"
+
+        taint=$(kubectl get nodes $node -o json | jq '.spec | select(.taints!=null) | .taints[] | select(.key=="kubemark") | select (.!=null) | select(.value=="true")' | wc -l)
+        if [ $taint -eq 0 ]; then
+          echo "Skipping $node, no 'kubemark' taint found"
+          continue
+        fi
+
+        status=$(kubectl get node $node -o json | jq '.status.conditions[] | select(.type=="Ready") | .status' --raw-output)
+        if [ $status != "Unknown" ]; then
```

Comment:
Why not invert it:
```
if [ $status == "Unknown" ]; then
        echo "Deleting node $node"
        kubectl delete node $node
fi
```

Reply:
I don't see any improvement in readibility compared to the current implementation. Can you more elaborate on that?